kernel/src/engine/arrow_utils.rs

```diff
@@ -20,10 +21,11 @@ use crate::arrow::array::{
 use crate::arrow::buffer::NullBuffer;
 use crate::arrow::compute::concat_batches;
 use crate::arrow::datatypes::{
-    DataType as ArrowDataType, Field as ArrowField, FieldRef as ArrowFieldRef, Fields,
+    DataType as ArrowDataType, Field as ArrowField, Field, FieldRef as ArrowFieldRef, Fields,
```

Comment:
oof, let's definitely not import `Field` as `ArrowField` AND `Field` :).  I'd prefer to stick with just `ArrowField`, but I guess as this PR shows that does have some developer overhead in that it's not obvious you need to use the mapped name.

Reply:
Oh yikes good catch! I think I let my LSP do all the work on this one 😓